src/transformers/generation/logits_process.py

```diff
@@ -1702,3 +1702,35 @@ def __call__(self, input_ids, scores):
         unconditional_logits = torch.nn.functional.log_softmax(logits[:, -1], dim=-1)
         out = self.guidance_scale * (scores - unconditional_logits) + unconditional_logits
         return out
+
+
+class EarlyStoppingLogitsProcessor(LogitsProcessor):
+    r"""This processor ensures that the EOS token is sampled if its probability is greater than the `min_eos_p`.
```

Comment:
I would add that this was created for BARK, in case someone stumbles upon this code